### PR TITLE
Delete deprecated bench-scripts from agent Makefile

### LINKED ISSUE #2801: [BUG] PR #2782 removed deprecated bench scripts but did not fix the agent Makefile to remove them
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

`agent/Makefile` still includes the deleted bench script.s

### To Reproduce

```
   cd agent/rpm
   make
```

### Actual Results

The result is

```
...
install --directory /tmp/opt/pbench-agent-0.72.0/agent/bench-scripts/templates
cd bench-scripts; \
    cp -a pbench-cyclictest pbench-dbench pbench-fio pbench-fio.md pbench-gen-iterations pbench-iozone pbench-linpack pbench-migrate pbench-netperf pbench-run-benchmark pbench-run-benchmark.pl pbench-run-benchmark-sample pbench-specjbb2005 pbench-uperf pbench-uperf.md pbench-user-benchmark templates /tmp/opt/pbench-agent-0.72.0/agent/bench-scripts
/home/nick/src/pbench/github/pbench/worktrees/python3-devel/agent/bench-scripts
cp: cannot stat 'pbench-cyclictest': No such file or directory
cp: cannot stat 'pbench-dbench': No such file or directory
cp: cannot stat 'pbench-iozone': No such file or directory
cp: cannot stat 'pbench-migrate': No such file or directory
cp: cannot stat 'pbench-netperf': No such file or directory
make[1]: *** [Makefile:276: install-bench-scripts] Error 1
make[1]: Leaving directory '/home/nick/src/pbench/github/pbench/worktrees/python3-devel/agent'
make: *** [../../utils/rpm.mk:68: agent] Error 2
...
```

### Expected Results

Successful creation of (s)rpm.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ndokos**
Fixes #2801

PR #2782 deleted deprecated bench scripts from the tree, but did
not delete them from the Makefile.

### CODE REVIEW

**Review by webbnh — APPROVED** [review]
If you don't mind a little feature creep, `pyproject.toml` needs updating, too, for a similar reason.  (The references to `binary-search.py` and `profile-builder.py` should be removed from the exclusions list, and `dbench-postprocess` should be removed from the inclusion list.)

**Review by dbutenhof — APPROVED** [review]

**Comment by ndokos** [comment]
@webbnh: I don't mind the feature creep but `dbench-postprocess` still exists in the tree (presumably it needs to go), so I presume I should delete it altogether (there is also a mocked version under `test-bin`).

**Review by portante — DISMISSED** [review]

**Comment by ndokos** [comment]
I pushed the changes above (and more) and I'll wait for @portante's approval.

**Review by dbutenhof — DISMISSED** [review]
🥳

File: agent/bench-scripts/postprocess/dbench-postprocess
Comment (webbnh) [inline agent/bench-scripts/postprocess/dbench-postprocess:1]: Does deleting this file require a change to `agent/Makefile` (e.g., line 91)?

**Review by webbnh — APPROVED** [review]
Yay!...but....

File: agent/bench-scripts/postprocess/dbench-postprocess
Comment (ndokos) [reply agent/bench-scripts/postprocess/dbench-postprocess:1]: No good deed goes unpunished :-(

Comment (ndokos) [reply agent/bench-scripts/postprocess/dbench-postprocess:1]: ... and I had to do this, otherwise my original problem would come back to bite me.

**Review by webbnh — APPROVED** [review]
🚀

**Comment by portante** [comment]
See also PR #2803.

**Review by portante — APPROVED** [review]

**Pull request merged**
